src/core/structures/PDFAcroForm.ts

```diff
@@ -22,12 +23,15 @@ class PDFAcroForm {
     this.dict = dict;
   }
 
-  Fields(): PDFArray {
-    return this.dict.lookup(PDFName.of('Fields'), PDFArray);
-  }
-
-  fieldDicts(): PDFDict[] {
-    return this.Fields().lookupElements(PDFDict);
+  Fields(): PDFAcroFormField[] {
+    const fieldDicts = this.dict
+      .lookup(PDFName.of('Fields'), PDFArray)
+      .lookupElements(PDFDict);
+    const fields = [];
+    for (let idx = 0, size = fieldDicts.length; idx < size; idx++) {
+      fields.push(PDFAcroFormField.fromDict(fieldDicts[idx]));
+    }
```

Comment:
Since we're going through a loop with a known size, it's more efficient to allocate the array up front and assign to indexes, rather than pushing onto it. This improves performance, which is the only reason to use `for` loops like this to begin with. Also (and this is pedantic), I've tried to standardize the `for` loop variables to `idx, len`, rather than `idx, size`. The choice is arbitrary, but I do think that consistency is valuable.

For example:
```js
const fields = new Array(fieldDicts.length);
for (let idx = 0, len = fieldDicts.length; idx < len; idx++) {
  fields[idx] = PDFAcroFormField.fromDict(fieldDicts[idx]);
}
```

**Sidenote:** https://github.com/Hopding/pdf-lib/issues/197 is a great example of just how impressively large some PDF files can really be!

Reply:
Updated